src/Estimators/src/SubModelKinDynWrapper.cpp

```diff
@@ -0,0 +1,273 @@
+                              logPrefix,
+                              subModel.getExternalContactList()[idx]);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool RDE::SubModelKinDynWrapper::inverseDynamics(Eigen::VectorXd& motorTorqueAfterGearbox,
+                                                 Eigen::VectorXd& frictionTorques,
+                                                 Eigen::VectorXd& tauExt,
+                                                 Eigen::VectorXd& baseAcceleration,
+                                                 Eigen::VectorXd& jointAcceleration)
+{
+    //    auto massMatrixEigen(iDynTree::toEigen(massMatrix));
+    long unsigned int nrOfJoints = subModel.getModel().getNrOfDOFs();
+    auto FTEigen(iDynTree::toEigen(massMatrix).block(6, 0, nrOfJoints, 6));
+    auto HEigen(iDynTree::toEigen(massMatrix).block(6, 6, nrOfJoints, nrOfJoints));
+
+    auto genBiasTorquesEigen(iDynTree::toEigen(genForces.jointTorques()));
+
+    auto pseudoInverseHEigen = HEigen.completeOrthogonalDecomposition().pseudoInverse();
+
+    jointAcceleration = pseudoInverseHEigen
+                        * (motorTorqueAfterGearbox - frictionTorques - genBiasTorquesEigen + tauExt
+                           + FTEigen * baseAcceleration);
+
+    return true;
+}
+
+bool RDE::SubModelKinDynWrapper::getBaseAcceleration(
+    std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel,
+    const Eigen::VectorXd& robotBaseAcceleration,
+    const Eigen::VectorXd& robotJointAcceleration,
+    Eigen::VectorXd& subModelBaseAcceleration)
+{
+    constexpr auto logPrefix = "[BipedalLocomotion::Estimators::SubModelKinDynWrapper::"
+                               "getBaseAcceleration]";
+
+    std::string subModelBaseFrameName = this->kinDyn.getFloatingBase();
+
+    iDynTree::Vector6 subModelBaseAccelerationIdyn;
+
+    if (!kinDynFullModel->getFrameAcc(subModelBaseFrameName,
+                                      iDynTree::make_span(robotBaseAcceleration.data(),
+                                                          robotBaseAcceleration.size()),
+                                      iDynTree::make_span(robotJointAcceleration.data(),
+                                                          robotJointAcceleration.size()),
+                                      subModelBaseAccelerationIdyn))
+    {
+        blf::log()->error("{} Unable to get the acceleration of the frame {}.",
+                          logPrefix,
+                          subModelBaseFrameName);
+        return false;
+    }
+
+    subModelBaseAcceleration = iDynTree::toEigen(subModelBaseAccelerationIdyn);
+
+    return true;
+}
+
+Eigen::VectorXd RDE::SubModelKinDynWrapper::getBaseVelocity(
+    std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel)
+{
+    std::string subModelBaseFrameName = this->kinDyn.getFloatingBase();
+
+    auto baseVelIdyn = kinDynFullModel->getFrameVel(subModelBaseFrameName);
+
+    return iDynTree::toEigen(baseVelIdyn);
```

Comment:
Note, here you are dynamically allocating memory every time you call this method. In fact, the ``toEigen`` method outputs a map. The map does not need memory, but since the output type is a ``VectorXd``, the compiler creates an intermediate object, allocating memory in the heap. This might be time-consuming. An alternative, is to return a const reference with the same type returned from ``getFrameVel``. Alternatively, you can add a ``VectorXd`` as class member, fill it in, and then return a const reference to it.